tasks/task04/collect_tree_leaves.py

```diff
@@ -0,0 +1,32 @@
+""""
+You are given a tree-like data structure represented as nested dictionaries.
+Implement a function collect_leaves that accepts a tree and returns a list of all its leaves. A leaf is a bottom-most node in a tree.
+
+Implement a kind of unit tests via assert operator.
+"""
+from typing import Union
+
+
+def collect_leaves(u: Union[dict, list]):
+    flatten_list = []
+    if isinstance(u, dict):
+        for item in u.values():
+            flatten_list.extend(collect_leaves(item))
+    else:
+        flatten_list.extend(u)

```

Comment:
you could use `return u` and it will be collected one level higher.

Reply:
reworked